### ISSUE TITLE
Bug: [no-unsafe-call] process.exit(1) is an "error" type typed value

### ISSUE THREAD

**Issue by CreativeTechGuy** [typescript-eslint/typescript-eslint#9625]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.3&fileType=.tsx&code=JYWwDg9gTgLgBAbzgUwB7HgXzgMyhEOAIgDsIATZALjHwGNkBnRogbgFgAoLtDACgCMASi5daEBswB0vGIKFA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tieQEMAZolpk%2B8eOiiJo0DtEjgwAXxBKgA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
process.exit(1)
```
```


### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-unsafe-call": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

I expected `process.exit(1)` to not be an error.

### Actual Result

There was an error when calling `process.exit(1)`

### Additional Info

I believe this error was introduced in #9304

**Comment by Josh-Cena** [comment]
Hmmm, fun. Are `never`-returning functions treated as error-typed?

**Comment by kirkwaiblinger** [comment]
Hi @CreativeTechGuy!

Does this repro for you locally or just in the playground? We've had some trouble with types in the playground lately, for example see 

> I believe this error was introduced in https://github.com/typescript-eslint/typescript-eslint/pull/9304

As also discussed in https://github.com/typescript-eslint/typescript-eslint/issues/9591, #9304 made _no_ changes to whether or not the rule reports - it just changes whether the error messages say "Unsafe call of an `error`-typed value", instead of "Unsafe call of an `any`-typed value"

**Comment by CreativeTechGuy** [comment]
Yes this repos locally which is how I found it when updating dependencies. So it's not just the playground @kirkwaiblinger

**Comment by kirkwaiblinger** [comment]
@Josh-Cena Looks like no; this doesn't flag....
```ts
function neverTypedFunction(): never {
  throw new Error();
}

neverTypedFunction(); 
```

**Comment by kirkwaiblinger** [comment]
@CreativeTechGuy 
Gotcha, thanks for the info! My guess is that it has something funky to do with how the types are configured in the tsconfig, but some investigation will be needed it seems.

**Comment by CreativeTechGuy** [comment]
I don't know the exact version this was introduced. But I can confirm that 7.14.1 was working. So something between 7.15.0-7.16.1 changed.

**Comment by kirkwaiblinger** [comment]
Just dropping one further breadcrumb for investigation, other functions on `process` that TS is ok with also run into this report, e.g. 
```ts
process.getActiveResourcesInfo()
```

**Comment by JoshuaKGoldberg** [comment]
Could someone post a standalone reproduction that isn't the playground please? We'll want to investigate this locally.

**Comment by bradzacher** [comment]
This is likely a local setup issue yeah. Need a repro to investigate further.

**Comment by CreativeTechGuy** [comment]
I'll work on a repro early next week. Thank you for looking into this!

**Comment by CreativeTechGuy** [comment]
It looks like it was a configuration issue. I had a global type augmentation for `process` so I could set the type of `process.env` in browser/bundler environments, and that was getting applied more broadly than I intended.

So I concede that it was user error and can be fixed. The error message was very confusing and for some reason didn't show up until I updated dependencies. Although I'm having a hard time finding a minimal reproduction.

Could the message "Unsafe call of an \`error\` type typed value" be renamed to something like "Unsafe call of an untyped value" or something like that? Something to indicate that it's an error because TS ESLint doesn't see it's type even though your editor/typescript does. That'd be helpful and point people in the direction of it being a configuration error with typescript configured differently than eslint.

**Comment by kirkwaiblinger** [comment]
@CreativeTechGuy 

Thanks for following up!

> Could the message be renamed?

Let's use https://github.com/typescript-eslint/typescript-eslint/issues/9591 for that discussion.

Cheers 🙂
